pkg/clients/common/secret.go

```diff
@@ -26,11 +27,34 @@ func (s *SuiteController) GetSecret(ns string, name string) (*corev1.Secret, err
 	return s.KubeInterface().CoreV1().Secrets(ns).Get(context.Background(), name, metav1.GetOptions{})
 }
 
-// Deleted a secret in a specified namespace
+// Delete a secret in a specified namespace
 func (s *SuiteController) DeleteSecret(ns string, name string) error {
 	return s.KubeInterface().CoreV1().Secrets(ns).Delete(context.Background(), name, metav1.DeleteOptions{})
 }
 
+// ListSecrets return a list of secrets from a namespace by label and selection limits
+func (s *SuiteController) ListSecrets(ns string, labelKey string, labelValue string, selectionLimit int64) (*corev1.SecretList, error) {
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{labelKey: labelValue}}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+		Limit:         selectionLimit,
+	}
+	return s.KubeInterface().CoreV1().Secrets(ns).List(context.Background(), listOptions)
+}
+
+// Delete all secrets in a specified namespace matching to label
+func (s *SuiteController) DeleteSecretsByLabel(ns string, labelKey string, labelValue string) error {
+	secretList, err := s.ListSecrets(ns, labelKey, labelValue, 1024)
+	if err != nil {
+		return err
+	}
+
+	for _, secret := range secretList.Items {
```

Comment:
perhaps we should at least log the secrets being deleted in case we reach the conclusion they shouldn't be there in the first place and want to circle back.

Reply:
Thank you! This is just a generic helper function, so it is printing/logging anything to behave similarly to other similar functions.

Regarding MPC, I think it is logging how it is deleting the secret like this (see last row):

```
2024-07-25T08:35:23Z    INFO    taskrun taskrun/taskrun.go:543  unassigning host build.appstudio.redhat.com/assigned-host from task     {"request": {"name":"test-rhta558bf2ac902920a3d3dd2dc636ccdc29-build-container-arm64","namespace":"test-rhtap-14-tenant"}}
2024-07-25T08:35:23Z    INFO    taskrun taskrun/dynamic.go:32   terminating cloud instances i-0b4f5650051e0e633 for TaskRun test-rhta558bf2ac902920a3d3dd2dc636ccdc29-build-container-arm64     {"request": {"name":"test-rhta558bf2ac902920a3d3dd2dc636ccdc29-build-container-arm64","namespace":"test-rhtap-14-tenant"}}
2024-07-25T08:35:23Z    INFO    taskrun aws/ec2.go:246  attempting to terminate AWS instance i-0b4f5650051e0e633        {"request": {"name":"test-rhta558bf2ac902920a3d3dd2dc636ccdc29-build-container-arm64","namespace":"test-rhtap-14-tenant"}}
2024-07-25T08:35:23Z    INFO    taskrun taskrun/taskrun.go:576  deleting secret from task       {"request": {"name":"test-rhta558bf2ac902920a3d3dd2dc636ccdc29-build-container-arm64","namespace":"test-rhtap-14-tenant"}}
```